### PR TITLE
V3 render primitive

### PR DESCRIPTION
**Pull request by dabingnn**
RenderPrimitive feature:
now it can work on ios, mac and android, windows and linux
Fast tmx has been migrated to renderPrimitive

### CODE REVIEW

File: cocos/renderer/CCPrimitive.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/****************************************************************************
+ Copyright (c) 2008-2010 Ricardo Quesada
+ Copyright (c) 2010-2012 cocos2d-x.org
+ Copyright (c) 2011      Zynga Inc.
+ Copyright (c) 2013-2014 Chukong Technologies Inc.
```
Comment (minggo) [inline cocos/renderer/CCPrimitive.cpp:2]: I think we should change these lines.
Just keep coco2d-x and chukong.

Comment (dabingnn) [inline cocos/renderer/CCPrimitive.cpp:2]: I copied the copyrights from other files.
If it is confirmed, I will change it.

File: cocos/renderer/CCPrimitive.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+class Primitive : public Ref
+{
+public:
+    static Primitive* create(VertexData* verts, IndexData* indices, PrimitiveType type);
+    
+    VertexData* getVertexData();
+    
+    IndexData* getIndexData();
```
Comment (minggo) [inline cocos/renderer/CCPrimitive.h]: Please add const.

File: cocos/2d/CCFastTMXLayer.cpp [hunk 3]
```diff
@@ -156,43 +153,47 @@ void TMXLayer::draw(Renderer *renderer, const Mat4& transform, uint32_t flags)
...
     int index = 0;
-    for(const auto& iter : _indicesVertexZNumber)
+//    for(const auto& iter : _indicesVertexZNumber)
+//    {
+//        auto& cmd = _renderCommands[index++];
+//        
+//        cmd.init(iter.first);
+//        cmd.func = CC_CALLBACK_0(TMXLayer::onDraw, this, _indicesVertexZOffsets[iter.first], iter.second);
+//        renderer->addCommand(&cmd);
+//    }
```
Comment (minggo) [inline cocos/2d/CCFastTMXLayer.cpp]: Please remove unneeded codes.

**Comment by minggo** [comment]
[ci rebuild]

**Comment by dabingnn** [comment]
[ci rebuild]

**Pull request merged**